packages/api/src/idpuser/validators.ts

```diff
@@ -0,0 +1,21 @@
+import Joi from "joi";
+import type { CreateTwoFactorMethodRequest } from "./models";
+
+export function validateCreateTwoFactorMethodRequest(
+  data: unknown
+): asserts data is CreateTwoFactorMethodRequest {
+  const validation = Joi.object()
+    .keys({
+      emailFromAuthToken: Joi.string().email().required(),
+      code: Joi.string().required(),
+      method: Joi.string().valid("email", "sms", "authenticator").required(),
```

Comment:
Authenticator is valid but we're not allowing it at the moment, so I would probably leave it out here.